Adds feature to customize searchable attributes on a resource

robinboening opened the pull request:
Example usage:

```
class Event < ActiveRecord::Base
  def self.searchable_alchemy_attributes
    %w(name date)
  end
end
```

Please see commit messages for further information.

TeatroIO commented:
I've prepared a stage to preview changes. <a href='http://customizable-searchable-attributes.alchemycms-alchemy-cms-42e8f4fc945384e0f6b5.ttrcloud.com/admin' target='_blank'>Open stage</a> or <a href='http://customizable-searchable-attributes.alchemycms-alchemy-cms-42e8f4fc945384e0f6b5.ttrcloud.com/admin/teatro.log' target='_blank'>view logs</a>.

tvdeyen commented on lib/alchemy/resource.rb:
```diff
@@ -159,19 +170,24 @@ def editable_attributes
+        model.searchable_alchemy_attributes
+      else
+        attributes.select { |a| searchable_attribute?(a) }
+          .concat(searchable_relation_attributes?(attributes))
```
Just recognized that we introduced a wrong method name in the last pull request `searchable_relation_attributes?` should not have a question mark in the the method name, because it does not return a Boolean, it returns a list of attributes instead. Would you mind to add a commit that fixes this? That would be great.

robinboening replied:
Sure. Will do in a second.

tvdeyen commented on lib/alchemy/resource.rb:
```diff
@@ -159,19 +170,24 @@ def editable_attributes
-        .concat(searchable_relation_attributes?(attributes))
+    def searchable_attribute_names
+      if model.respond_to?(:searchable_alchemy_attributes)
+        model.searchable_alchemy_attributes
```
To be consistent to all other methods of this like, we should rename this to `searchable_alchemy_resource_attributes`. This is longer, but reflects more what it's for: Alchemy resources and it follows the naming of the other methods.

robinboening replied:
Agreed.

tvdeyen commented:
@robinboening Great feature. Thanks. Could you please review the comments above and add a changelog entry? Thanks

robinboening commented:
- [x] Renamed `searchable_relation_attributes?` to `searchable_relation_attributes` in a separate commit
- [x] Renamed `searchable_alchemy_attributes` to `searchable_alchemy_resource_attributes` to follow the previous naming of the public API. (Amended to b739cf099d09c01cb829bef3cf4620bfb16db6df and force pushed)

tvdeyen commented:
@robinboening Please add a changelog entry. Thanks

robinboening commented:
Done! See 803108049095dbfe7aeb893d90e3655211c7575b

tvdeyen commented:
Thanks! 🍰 🍨
